hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-jobclient/src/test/java/org/apache/hadoop/mapreduce/TestNewCombinerGrouping.java

```diff
@@ -174,6 +176,9 @@ public void testCombiner() throws Exception {
     } else {
       Assert.fail("Job failed");
     }
+    if (testDir.exists()) {
```

Comment:
this doesn't get invoked if the job failed, and as there's no cleanup before L109, no recovery.